java/ql/automodel/src/AutomodelFrameworkModeCharacteristics.qll

```diff
@@ -272,76 +383,5 @@ private class NonPublicMethodCharacteristic extends CharacteristicsImpl::Uninter
-  boolean ignoreKnownModelingErrors
-) {
-  // An endpoint's characteristics should not include positive indicators with medium/high confidence for more than one
-  // sink/source type (including the negative type).
-  exists(
-    EndpointCharacteristic characteristic2, AutomodelEndpointTypes::EndpointType endpointClass2,
-    float confidence2
-  |
-    endpointType != endpointClass2 and
-    (
-      endpointType instanceof AutomodelEndpointTypes::SinkType and
-      endpointClass2 instanceof AutomodelEndpointTypes::SinkType
-      or
-      endpointType instanceof AutomodelEndpointTypes::SourceType and
-      endpointClass2 instanceof AutomodelEndpointTypes::SourceType
-    ) and
-    characteristic.appliesToEndpoint(endpoint) and
-    characteristic2.appliesToEndpoint(endpoint) and
-    characteristic.hasImplications(endpointType, true, confidence) and
-    characteristic2.hasImplications(endpointClass2, true, confidence2) and
-    confidence > SharedCharacteristics::mediumConfidence() and
-    confidence2 > SharedCharacteristics::mediumConfidence() and
-    (
-      ignoreKnownModelingErrors = true and
-      not knownOverlappingCharacteristics(characteristic, characteristic2)
-      or
-      ignoreKnownModelingErrors = false
-    )
-  ) and
-  errorMessage = "Endpoint has high-confidence positive indicators for multiple classes"
-  or
-  // An endpoint's characteristics should not include positive indicators with medium/high confidence for some class and
-  // also include negative indicators with medium/high confidence for this same class.
-  exists(EndpointCharacteristic characteristic2, float confidence2 |
-    characteristic.appliesToEndpoint(endpoint) and
-    characteristic2.appliesToEndpoint(endpoint) and
-    characteristic.hasImplications(endpointType, true, confidence) and
-    characteristic2.hasImplications(endpointType, false, confidence2) and
-    confidence > SharedCharacteristics::mediumConfidence() and
-    confidence2 > SharedCharacteristics::mediumConfidence()
-  ) and
-  ignoreKnownModelingErrors = false and
-  errorMessage = "Endpoint has high-confidence positive and negative indicators for the same class"
-}
-
-/**
- * Holds if `characteristic1` and `characteristic2` are among the pairs of currently known positive characteristics that
- * have some overlap in their results. This indicates a problem with the underlying Java modeling. Specifically,
- * `PathCreation` is prone to FPs.
- */
-private predicate knownOverlappingCharacteristics(
-  EndpointCharacteristic characteristic1, EndpointCharacteristic characteristic2
-) {
-  characteristic1 != characteristic2 and
-  characteristic1 = ["mad taint step", "create path", "read file", "known non-sink"] and
-  characteristic2 = ["mad taint step", "create path", "read file", "known non-sink"]
+  override predicate appliesToEndpoint(Endpoint e) { not e.getEnclosingCallable().isPublic() }
```

Comment:
I'm slightly confused about the interplay between this an the use of `ModelExclusions::ModelApi` elsewhere (e.g. the IPA type for sources, and `NotAModelApi`). I feel like we're checking `isPublic` in more than one place (though I guess not everything is assumed to be an instance of `ModelApi`).

Reply:
Dropping this, iow, not adding this characteristic on top would reflect the view that the language team (maintainers of `ModelApi`) are the real experts for what is worth having a model for.

Reply:
👀 @jhelie

Reply:
I would be in favour of dropping it. 👍

Reply:
I would be in favour of dropping it. 👍

Reply:
It doesn't sound like something that should make up for a huge increase. Some data points:

`apache/geode`: 103625 candidates ➡️ 114467 candidates (+10%)
`square/okhttp`:  11841 candidates ➡️ 11867 candidates (+0,2%)

Sounds good, @jhelie?

Reply:
Btw, getting more candidates is a _good_ thing, it gives us more complete models 🙂 👍

Reply:
Yes I think we can live with a 10% increase 👍 

> Btw, getting more candidates is a good thing, it gives us more complete models 🙂 👍

Agreed as long as it's not +10000%, we're hammering the API enough as it is 😅

Reply:
haha, yes. that would raise some concern, agreed 👍

Reply:
[fdbc553](https://github.com/github/codeql/pull/14197/commits/fdbc55302933c7c56f42ce19cc038dc294744496) ✅